File: credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long number = get_long("Number: ");
    long number2 = number;
    int length = 2;
    
    while (number2 > 100)
    {
        number2 = number2 / 10;
        length = length + 1;
    }
    
    int luhn = 0;
    int i = 1;
    
    while (number)
    {
        if (i % 2 == 1)
        {
            luhn += number % 10;
            number = number / 10;
        }
        
        else
        {
            luhn += ((number % 10) * 2) / 10;
            luhn += ((number % 10) * 2) % 10;
            number = number / 10;
        }
        
        i++;
    }
    
    if (luhn % 10 == 0)
    {
        if ((number2 == 34 || number2 == 37) && (length == 15))    
        {
            printf("AMEX\n");
        }
        
        else if ((number2 > 50 && number2 < 56) && (length == 16))   
        {
            printf("MASTERCARD\n");
        }
        
        else if ((number2 / 10 == 4) && (length == 13 || length == 16)) 
        {
            printf("VISA\n");
        }
        
        else
        {
            printf("INVALID\n");
        }
    }
    
    else
    {
        printf("INVALID\n");
    }
}

File: credit.py
from cs50 import get_int

number = get_int("Number: ")
number2 = number
length = 2
    
while True:
    number2 = int(number2 / 10)
    length = length + 1
    if number2 < 100:
        break
    
luhn = 0
i = 1
    
while True:
    if i % 2 == 1:
        luhn += number % 10
        number = int(number / 10)
        
    else:
        luhn += int(((number % 10) * 2) / 10)
        luhn += ((number % 10) * 2) % 10
        number = int(number / 10)
        
    i += 1
    
    if number == 0:
        break
    
if luhn % 10 == 0:
    if (number2 == 34 or number2 == 37) and (length == 15):
        print("AMEX")
        
    elif (number2 > 50 and number2 < 56) and (length == 16):
        print("MASTERCARD")
        
    elif (number2 / 10 == 4) and (length == 13 or length == 16):
        print("VISA")
        
    else:
        print("INVALID")
    
else:
    print("INVALID")